### PR TITLE
Add `nester log`

### PR DESCRIPTION
**Pull request by ByteOtter**
# What does this PR change?

This PR consists of two components:

1. `nester log` module which logs which projects in which languages have been created when and where with automatic tracking when projects have been deleted using `nester clean <PROJECTNAME>`. (Logging is optional and can be toggled using the `--no-log`-flag)
2. The `nester log` command which is supposed to print out the existing projects, their location and the languages they have been written in as a table to the console.

|Task|status|todo|
|-|-|-|
|Implement Feature| Done |x|
|Change Docs | Done |x|
|Implement Tests| Doing | Unittests and Integration tests for this feature |
|Satisfy workflows| Done| x |

Tick the applicable box:
- [x] Add new feature
- [ ] Add language support
- [ ] UI improvement
- [ ] Security changes
- [x] Tests
- [x] Documentation changed
<br/>

- [ ] General Maintenance

## UI changes

- before: Nester's console output was uniform making it hard to spot errors

- after: Some color has been added to certain events depending on success, info or failure.

 - [x] DONE

## Links

Fixes: #34 

- [ ] DONE

## Documentation

- `usage_guide.rst` Updated and built successfully locally.

- [x] DONE

### CODE REVIEW

File: src/nester/nester_log.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+            for line in log_file:
+                project_name = re.search(r"[-\s]+Project:\s+(.*?)\s+-", line).group(1)
+                language = re.search(r"[-\s]+Language:\s+(.*?)\s+-", line).group(1)
+                location = re.search(r"[-\s]+Location:\s+(.*)", line).group(1)
+
+                print(f"{project_name:<20} | {language:<10} | {location}")
+            print()
+    except FileNotFoundError:
+        print("\033[31FATAL ERROR -- LOG FILE NOT FOUND - ABORTING!\033[0m")
```
Comment (tiltingpenguin) [inline src/nester/nester_log.py:113]: Nitpick: We should use the same error message for all FileNotFound errors. I think the other one in remove_log_entry fits better

File: src/nester/nester_log.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        record.projectname = getattr(record, "projectname", "N/A")
+        record.programming_language = getattr(record, "programming_language", "N/A")
+        record.location = getattr(record, "location", "N/A")
+        return record
+
+
+logger = logging.getLogger("nester")
+logger.setLevel(logging.INFO)
+
+log_file_path = Path.home() / "nester.log"
```
Comment (tiltingpenguin) [inline src/nester/nester_log.py:30]: I think that if we go this way and write into a file in the home directory we should at least make it a hidden file

**Review by tiltingpenguin — DISMISSED** [review]
As mentioned before We need some error handling in case we do not have a log entry but the project still exists. Also please add tests before merging

File: src/nester/nester_log.py (hunk 2 above)
Comment (ByteOtter) [reply src/nester/nester_log.py]: Agreed, Fixed

**Pull request merged**